server/src/test/java/org/elasticsearch/indices/state/CloseIndexIT.java

```diff
@@ -0,0 +1,343 @@
+                                IndexResponse response = client().prepareIndex(indexName, "_doc").setSource("num", randomInt()).get();
+                                if (response.status() == RestStatus.CREATED) {
+                                    acknowledgedDocs.incrementAndGet();
+                                }
+                            } catch (Exception e) {
+                                assertException(e, indexName);
+                            }
+                            break;
+                        }
+                        case 1: // Bulk docs indexing
+                        {
+                            BulkRequestBuilder request = client().prepareBulk(indexName, "_doc");
+                            for (int j = 0; j < randomIntBetween(1, 10); j++) {
+                                request.add(new IndexRequest().source("num", randomInt()));
+                            }
+
+                            BulkResponse response = request.get();
+                            startClosing.countDown();
+                            for (BulkItemResponse itemResponse : response) {
+                                if (itemResponse.isFailed() == false) {
+                                    acknowledgedDocs.incrementAndGet();
+                                } else {
+                                    assertException(itemResponse.getFailure().getCause(), indexName);
+                                }
+                            }
+                            break;
+                        }
+                        case 2: // Single doc update
+                        {
+                            try {
+                                int docId = randomIntBetween(0, nbDocs - 1);
+                                client().prepareUpdate(indexName, "_doc", String.valueOf(docId)).setDoc("num", randomInt()).get();
+                            } catch (VersionConflictEngineException e) {
+                                // it's ok to ignore
+                            } catch (Exception e) {
+                                assertException(e, indexName);
+                            }
+                            break;
+                        }
+                        default:
+                            throw new AssertionError("Illegal randomisation branch");
+                    }
+                }
+            });
+            threads[i].start();
+        }
+
+        try {
+            startIndexing.countDown();
+            startClosing.await();
+            assertAcked(client().admin().indices().prepareClose(indexName));
+        } finally {
+            runIndexing.set(false);
+        }
+
+        for (Thread thread : threads) {
+            thread.join();
+        }
+
+        assertIndexIsClosed(indexName);
+        assertAcked(client().admin().indices().prepareOpen(indexName));
+        assertHitCount(client().prepareSearch(indexName).setSize(0).setFetchSource(false).get(), acknowledgedDocs.get());
```

Comment:
why call `setFetchSource` if size is `0`?

Reply:
This is useless, indeed.